### ISSUE TITLE
Issue #4362 still happening in Android SDK 4.0.0

### ISSUE THREAD

**Issue by roger-schaer** [mapbox/mapbox-gl-native#4764]
Hello, 

I still have the issue described in #4362 and don't know how to proceed:

I followed the examples described in :
* https://www.mapbox.com/android-sdk/offline-map/
* https://www.mapbox.com/android-sdk/offline-manager/

The only difference being that my OfflineManager is in a different Activity than the MapView.

But I cannot get the map to show when no network connection is available. I tried both with my own style URL and the default Emerald style, for example.

Once the phone is set to flight mode (or simply the WiFi and Mobile Data are turned off) and the app is opened, the OfflineManager detects that the region has been completely downloaded (through a OfflineRegion.OfflineRegionStatusCallback), but it is not able to load the data, and I am seeing several entries in logcat about trying to contact the mapbox API server:

```
{Map}[Setup]: loading style failed: Unable to resolve host "api.mapbox.com": No address associated with hostname
{Map}[Style]: Failed to load sprite: Unable to resolve host "api.mapbox.com": No address associated with hostname
{Map}[Style]: Failed to load tile 0/0/0 for source mapbox: Unable to resolve host "api.mapbox.com": No address associated with hostname
```

I also tried with the MapBox SDK 4.1-SNAPSHOT, but that one gives me an OutOfMemoryError while downloading the map.

I finally also copy-pasted the example from https://www.mapbox.com/android-sdk/offline-manager/ into my app, and it is crashing without a Network connection, because it is unable to run the "getMapAsync" method it seems (the onMapReady is never called).

Please could you point me in the right direction? It is the first time I am asking for help on GitHub, so I apologize if I didn't respect some ground rules about submitting a bug.

Thank you very much in advance for any help you could provide me!

**Platform:** Android
**Mapbox SDK version:** 4.0.0

### Steps to trigger behavior

 1. Download a region using the OfflineManager
 2. Close the app, turn off the WiFi
 3. Reopen the app

### Expected behavior
Map should be displayed

### Actual behavior
Map is not displayed, and logcat shows 'Unable to resolve host "api.mapbox.com": No address associated with hostname'

**Comment by jfirebaugh** [comment]
Are you certain that when you launch the app without network connectivity, you are viewing an area on the map that is included in the offline region? Note that the map must be geographically within the offline region's bounding box, and the map zoom must be within the zoom range of the offline region.

**Comment by roger-schaer** [comment]
Thank you for the prompt response!
Indeed, it seems the map is not centered on the offline area when I start the Activity, but in this case I have the problem that to move the camera there, I need to access the "map" object that should be set when "getMapAsync" completes (if I understood the code correctly), which it never does in my case.
I would be really grateful for any pointers you could give me about this!

**Comment by jfirebaugh** [comment]
cc @tobrun

**Comment by tobrun** [comment]
@roger-schaer 

There are a couple of ways to initialise a map early with a configuration. This was setup so you are able to configure your map early and do things before the `getMapAsync` completes. This allows you to provide a better UX for your users. 

How this is setup depends on which component you use from the SDK:
####  MapView in xml
 - Exposes custom xml attributes as `center_longitude`, `center_latitude` and `zoom`
 
#### MapView in code
 - Exposes a constructor that take in a MapboxMapOptions object (see Fragment above)

#### MapFragment
  - Exposes a `newInstance(MapboxMapOptions options)` method. The MapboxMapOptions class exposes a method called `camera(CameraPosition cameraPosition)`.

Reach out if you have any more questions and LMK if this resolves your problem!

**Comment by roger-schaer** [comment]
Thank you very much @tobrun!

Configuring the map early using the custom XML attributes in the layout file solved my problem, both with integrated styles as well as my custom style, so this is perfect!

Maybe one more question if I may : I am now trying to optimize the size of the map. I set zoom levels between 14 and 18 on a ~1 square kilometer area in a city in Switzerland (the tile estimator indicates 20 tiles). Using the default "Light" Mapbox style, it seems it is downloading at least 100MB of data. Is this to be expected? It seems a little bit large for a relatively small area.

Thank you again for your help, I am really impressed by the features of Mapbox and Mapbox studio!

I will close the issue, as the original problem was solved already.

**Comment by tobrun** [comment]
@roger-schaer

We aware that current downloaded regions are a bit on the large size.  
This is definitely we want to improve in future releases but note that first iteration of offline functionality just landed with the `4.0.0` release. 
Thank you for providing feedback and please reach out if you have any other questions!
